src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
-                        typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        nameof(IConfigurationBuilder),
+                        nameof(IFileProvider)));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
+            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(fileProvider.GetFileInfo("/").PhysicalPath));
```

Comment:
Or you add an Optional parameter like the other providers use: https://github.com/aspnet/UserSecrets/issues/66